### PR TITLE
Refactor processCitation in GrobidService to match processPdf

### PR DESCRIPTION
**Pull request by btut**
GrobidService.processCitation now returns an Optional<BibEntry> to match the signature of processPDF.
This also makes the tests more expressive, as we can compare the BibEntry to an expected BibEntry instead of comparing Strings.

- [] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [x] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [] Screenshots added in PR description (for UI changes)
- [] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

### CODE REVIEW

**Review by Siedlerchr — APPROVED** [review]
At least better than assertTrue. I would prefer if the response deliverd by the processCitaiton would be directly a BibEntry instead of a string, so you could compare two bib entries:

**Comment by btut** [comment]
> At least better than assertTrue. I would prefer if the response deliverd by the processCitaiton would be directly a BibEntry instead of a string, so you could compare two bib entries:

I agree that would make it more consistent with processPDF, so I changed it.

File: src/test/java/org/jabref/logic/importer/util/GrobidServiceTest.java [hunk 1]
```diff
@@ -38,37 +40,36 @@ public static void setup() {
...
     }
 
     @Test
-    public void processEmptyStringTest() throws IOException {
-        String response = grobidService.processCitation(" ", GrobidService.ConsolidateCitations.WITH_METADATA);
+    public void processEmptyStringTest() throws IOException, ParseException {
+        Optional<BibEntry> response = grobidService.processCitation(" ", importFormatPreferences, GrobidService.ConsolidateCitations.WITH_METADATA);
         assertNotNull(response);
-        assertEquals("", response);
+        assertTrue(response.isEmpty());
```
Comment (Siedlerchr) [inline src/test/java/org/jabref/logic/importer/util/GrobidServiceTest.java:67]: assertEquals(Optiona.empty(), response)

File: src/test/java/org/jabref/logic/importer/util/GrobidServiceTest.java [hunk 2]
```diff
@@ -12,16 +12,17 @@
...
 import org.jabref.model.entry.BibEntry;
 import org.jabref.model.entry.field.StandardField;
+import org.jabref.model.entry.types.StandardEntryType;
 import org.jabref.testutils.category.FetcherTest;
 
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
 import org.mockito.Answers;
 
+import static org.junit.Assert.assertTrue;
```
Comment (Siedlerchr) [inline src/test/java/org/jabref/logic/importer/util/GrobidServiceTest.java:22]: You have the wrong junit, it. must be import static org.junit.jupiter.api.Assertions.assertTrue;

**Review by Siedlerchr — APPROVED** [review]

**Comment by Siedlerchr** [comment]
Much better now! :)

**Pull request merged**